### PR TITLE
#30 restructure app folders, add breadcrumbs

### PR DESCRIPTION
**Pull request by WitchDevelops**
- **Restructured the folders a bit, to have a better idea how the app will be structured**

Now the main view with the pet grid has a route `/user` - it will be `/[user-id]` once we set up authentication
(the main page at `/` is a mock, will rework that on another branch)

- **Added breadcrumbs component and added them to the main layout**

Breadcrumbs have hardcoded links for now. Cannot implement dynamic links without proper routing in place, and routing is kinda blocked by the authenthication (`/user/[pet-id]`)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,8 +20,8 @@
     "@radix-ui/react-select": "^2.0.0",
     "@radix-ui/react-slot": "^1.1.0",
     "@radix-ui/react-switch": "^1.0.3",
-    "@radix-ui/react-toast": "^1.1.5",
     "@radix-ui/react-tabs": "^1.1.0",
+    "@radix-ui/react-toast": "^1.1.5",
```
Comment (WitchDevelops) [inline package.json:24]: artifact after merging main

File: package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
     "react-dom": "^18",
     "react-hook-form": "^7.52.0",
     "react-icons": "^5.2.1",
+    "sass": "^1.77.6",
```
Comment (WitchDevelops) [inline package.json:35]: installed SASS for more custom styling, now if you want to write custom styles go to `/app/styles/`, create a `_[page-name].scss` and import it into `main.scss`

File: src/app/page.tsx
Comment (WitchDevelops) [inline src/app/page.tsx:1]: This page will be the home page the user sees before logging in.
Now it's the door to Moria ;D

File: src/app/user/layout.tsx
Comment (WitchDevelops) [inline src/app/user/layout.tsx:1]: This will serve as the layout for the main page a user sees after logging in.

File: src/components/BreadCrumbs.tsx
Comment (WitchDevelops) [inline src/components/BreadCrumbs.tsx:1]: run `npm install` because those components were installed from Shadcn

**Review by jagodakubicka — APPROVED** [review]

**Pull request merged**